save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/SettingsView.kt

```diff
@@ -66,14 +67,22 @@ typealias FieldsStateSetter = StateSetter<SettingsInputFields>
 @Suppress("MISSING_KDOC_CLASS_ELEMENTS")
 external interface SettingsProps : PropsWithChildren {
     /**
-     * Currently logged in user or null
+     * Currently logged-in user or null
      */
     var userInfo: UserInfo?
 
     /**
      * just a flag for a factory
      */
     var type: FrontendRoutes
+
+    /**
+     * After updating user information we will update userSettings without re-rendering the page
+     * PLEASE NOTE: THIS PROPERTY AFFECTS RENDERING OF WHOLE APP.KT
+     * IF YOU HAVE SOME PROBLEMS WITH IT, CHECK THAT YOU HAVE PROPAGATED IT PROPERLY:
+     * { this.userInfoSetter = (!) PROPS (!) .userInfoSetter }
+     */
+    var userInfoSetter: StateSetter<UserInfo?>
```

Comment:
It also might be a good idea to pass a callback that triggers user info fetch instead of passing a `StateSetter`

Reply:
I actually don't get this suggestion 😄  - to my mind it is quite clear to propagate setter as a callback and update userInfo only on the high level of App.kt

Can you pls clarify?

Reply:
My idea is to fetch `UserInfo` from backend instead of updating `UserInfo` that was once fetched. We could create some `fetchUserInfo` callback that would fetch the data and update the `userInfo` that is passed everywhere. That might be a better idea sometimes